dask_planner/src/sql/logical.rs

```diff
@@ -140,19 +143,15 @@ impl PyLogicalPlan {
     }
 
     #[pyo3(name = "getRowType")]
-    pub fn row_type(&self) -> RelDataType {
-        let fields: &Vec<DFField> = self.original_plan.schema().fields();
-        let mut rel_fields: Vec<RelDataTypeField> = Vec::new();
-        for i in 0..fields.len() {
-            rel_fields.push(
-                RelDataTypeField::from(
-                    fields[i].clone(),
-                    self.original_plan.schema().as_ref().clone(),
-                )
-                .unwrap(),
-            );
-        }
-        RelDataType::new(false, rel_fields)
+    pub fn row_type(&self) -> PyResult<RelDataType> {
```

Comment:
PyO3 will automatically wrap any return type from an annotated function with `PyResult<T>` but maybe it is best to explicitly lay that out. I like that better too.